### PR TITLE
Contributer history

### PR DESCRIPTION
**Pull request by kevincharles**

### CODE REVIEW

File: public/api/duplicatePortfolioActivity.php [hunk 1]
```diff
@@ -199,8 +202,68 @@
 
     // TODO: how do we handle support files?
 
+
+    //content_contributor_history
+    //Don't update history if duplicating to the same course
+    if ($nextCourseId != $prevCourseId){
+    $insert_content_contributor_history = [];
```
Comment (jaltekruse) [inline public/api/duplicatePortfolioActivity.php:209]: Missing indentation

File: public/api/getPortfolioActivityView.php [hunk 2]
```diff
@@ -46,40 +51,54 @@
...
-        if ($result->num_rows > 0) {
-            $row = $result->fetch_assoc();
-            $contributor['firstName'] = $row['firstName'];
-            $contributor['lastName'] = $row['lastName'];
-            $contributor['profilePicture'] = $row['profilePicture'];
-        }
+    if ($result->num_rows > 0) {
+        while($row = $result->fetch_assoc()){
+            array_push($contributors, [
+                "courseId" => $row["courseId"],
```
Comment (jaltekruse) [inline public/api/getPortfolioActivityView.php:89]: fetch_assoc() already gives an associative array, we really don't have to keep doing this. You can control what you want out of the table using just the select list in the SQL.

File: public/api/duplicatePortfolioActivity.php (hunk 1 above)
Comment (kevincharles) [reply public/api/duplicatePortfolioActivity.php:209]: Done

File: public/api/getPortfolioActivityView.php [hunk 3]
```diff
@@ -8,69 +8,117 @@
...
-    $json = json_decode($row["json"], true);
-}
+    if ($result->num_rows > 0) {
+        $row = $result->fetch_assoc();
+        $isBanned = $row['isBanned'];
+        if ($isBanned == '1'){
+            throw new Exception("Activity has been banned.");
+        }
+        $label = $row['label'];
+        $json = json_decode($row["json"], true);
```
Comment (jaltekruse) [inline public/api/getPortfolioActivityView.php:48]: It doesn't look like we are doing anything with this json, we are just reading it out of the database, decoding it into this variable and then dropping it.

**Review by jaltekruse — APPROVED** [review]
+1

**Pull request merged**